addon/-private/fields/key.ts

```diff
@@ -9,7 +9,9 @@ export default function key(options?: KeyDefinition);
 export default function key(options: Model | KeyDefinition = {}, _?: unknown) {
   function trackedKey(target: any, property: string, _: PropertyDescriptor) {
     function get(this: Model) {
-      return getKeyCache(this, property).value;
```

Comment:
Can you add an assertion here as well please?

Reply:
Sure thing, sorry about that. Added!